shared-bindings/storage/__init__.c

```diff
@@ -160,16 +161,38 @@ MP_DEFINE_CONST_FUN_OBJ_1(storage_getmount_obj, storage_getmount);
 //|     This function can be called from the REPL when ``CIRCUITPY``
 //|     has become corrupted.
 //|
+//|     :param bool extended: On boards that support ``dualbank`` module
+//|         and the ``extended`` parameter, the ``CIRCUITPY`` storage can be
+//|         extended by setting this to `True`. If this isn't provided or
+//|         set to `None` (default), the existing configuration will be used.
+//|
 //|     .. warning:: All the data on ``CIRCUITPY`` will be lost, and
 //|          CircuitPython will restart on certain boards."""
 //|     ...
 //|
 
-STATIC mp_obj_t storage_erase_filesystem(void) {
-    common_hal_storage_erase_filesystem();
+STATIC mp_obj_t storage_erase_filesystem(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_extended };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_extended, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+    };
+
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    #if CIRCUITPY_STORAGE_EXTEND
+    bool extended = (args[ARG_extended].u_obj == mp_const_none) ? supervisor_flash_get_extended() : mp_obj_is_true(args[ARG_extended].u_obj);
+    common_hal_storage_erase_filesystem(extended);
+    #else
+    if (args[ARG_extended].u_obj != mp_const_none) {
+        mp_raise_ValueError_varg(translate("%q isn't supported"), MP_QSTR_extended);
```

Comment:
Use NotImplementedError instead, and then we can also avoid adding a new string. (You'll need to `make translate again).
```suggestion
        mp_raise_NotImplementedError_varg(translate("%q"), MP_QSTR_extended);
```

Reply:
How about the following:
```c
if (args[ARG_extended].u_obj != mp_const_none && mp_obj_is_true(args[ARG_extended].u_obj)) {
    mp_raise_NotImplementedError(translate("%q=%q"), MP_QSTR_extended, MP_QSTR_True);
}
```

Reply:
that's fine too

Reply:
and even clearer